src/middlewared/middlewared/plugins/activedirectory.py

```diff
@@ -1181,7 +1181,16 @@ def validate_domain(self, data=None):
 
         netlogon_ping = await run([SMBCmd.WBINFO.value, '-P'], check=False)
         if netlogon_ping.returncode != 0:
-            raise CallError(netlogon_ping.stderr.decode().strip('\n'))
+            wberr = netlogon_ping.stderr.decode().strip('\n')
+            err = errno.EFAULT
+            for wb in WBCErr:
+                if wb.err() in wberr:
+                    humanized_text = wb.value[0]
+                    wberr.replace(wb.err(), humanized_text)
```

Comment:
Did you mean `wberr = wberr.replace(wb.err(), humanized_text)` ?

Reply:
You caught me before I finished refactoring :)